src/pulsar_producer.erl

```diff
@@ -871,8 +932,9 @@ do_collect_send_requests(Acc, Count, Limit) ->
 
 try_close_socket(#{sock := undefined}) ->
     ok;
-try_close_socket(#{sock := Sock, opts := Opts}) ->
+try_close_socket(#{sock := Sock, sock_pid := SockPid, opts := Opts}) ->
     _ = pulsar_socket:close(Sock, Opts),
+    ok = pulsar_socket_writer:stop(SockPid),
```

Comment:
nit: maybe add a comment here to record the importance of closing socket before stopping the writer process.
(because the writer process might be blocked at writing, closing the socket should cause it to stop immediately with `einval` error)

Reply:
:checkered_flag: